### ISSUE TITLE
Braking Execution of a Template

### ISSUE THREAD

**Issue by endeffects** [lunet-io/scriban#182]
I tried to use your "ret" command to stop the execution of a template if a pre condition is not filled.
But i've noticed that this effects other templates too, so they also got not rendered, still if they don't have such pre conditions. Is there any better option to exit the template execution?

```
   {{~ if !Settings.FlagA || !Settings.FlagB; ret; end; ~}}
```

This works well, if i inverse the condition, but that is not my prefered solution:

```
   {{~ if Settings.FlagA && Settings.FlagB ~}}
   ...
   {{~ end ~}}
```

I also tried this, with the same strange result:

```
   {{~ if !Settings.FlagA || !Settings.FlagB ~}}
   {{~ ret ~}}
   {{~ end ~}}
```

**Comment by xoofx** [comment]
>  that this effects other templates too,

Sorry, I'm not sure to understand that. What are these other templates? How they are used?

**Comment by endeffects** [comment]
i'm rendering 100s of templates per task, 
when i use the ret command in some of them,
many templates got not rendered,
still if they don't have the ret command.

**Comment by xoofx** [comment]
hm, are you using the same `TemplateContext` ?

**Comment by endeffects** [comment]
yes

**Comment by xoofx** [comment]
I haven't checked but a quick glance at the code and it is likely that the state is not restored. It requires a fix. As a temporary workaround, the only way is to use a new TemplateContext every time.

**Comment by endeffects** [comment]
thank you :)

**Comment by xoofx** [comment]
Fixed in 2.1.3 coming later today.
